### PR TITLE
Optimizing fp8_fp16 mixedprec gemm

### PR DESCRIPTION
**Pull request by zjing14**

### CODE REVIEW

File: example/35_splitK_gemm/splitK_gemm_xdl_fp16_fp8.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline example/35_splitK_gemm/splitK_gemm_xdl_fp16_fp8.cpp]:
```suggestion
// Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
```

File: include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp [hunk 2]
```diff
@@ -21,50 +21,11 @@ struct PassThroughPack2
...
-        uint16_t t = ck::bit_cast<uint32_t>(x);
-        y          = ck::bit_cast<ck::f8x2_t>(t);
-    }
-
     __host__ __device__ constexpr void operator()(ck::half2_t& y, const ck::f8x2_t& x) const
     {
         auto t = type_convert<float2_t>(x);
         y      = type_convert<half2_t>(t);
     }
-
```
Comment (bartekxk) [inline include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp:36]: Is it not needed anymore?

File: include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp [hunk 3]
```diff
@@ -1156,27 +1158,58 @@ struct ThreadwiseTensorSliceTransfer_v4
...
-                constexpr index_t dst_offset = dst_desc.CalculateOffset(
-                    dst_origin_idx + data_to_origin_disp_idx + i * src_scalar_step_in_vector);
+                constexpr index_t pack_size = 2;
 
-                dst_buf(Number<dst_offset>{}) = dst_tmp_vector.template AsType<DstData>()[i];
-            });
+                using dst_v_t = typename vector_type_maker_t<DstData, pack_size>::type;
+                using src_v_t = typename vector_type_maker_t<SrcData, pack_size>::type;
+
+                // TODO: if SrcData and DstData are vetor type, then static_cast may not compile
```
Comment (bartekxk) [inline include/ck/tensor_operation/gpu/thread/threadwise_tensor_slice_transfer.hpp]:
```suggestion

```

File: library/src/tensor_operation_instance/gpu/gemm_splitk/device_gemm_xdl_splitk_f16_fp8_f16_mk_nk_mn_kpb128_instance.cpp [hunk 4]
```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2024, Advanced Micro Devices, Inc. All rights reserved.
```
Comment (bartekxk) [inline library/src/tensor_operation_instance/gpu/gemm_splitk/device_gemm_xdl_splitk_f16_fp8_f16_mk_nk_mn_kpb128_instance.cpp]:
```suggestion
// Copyright (c) 2024, Advanced Micro Devices, Inc. All rights reserved.
```

**Review by bartekxk — DISMISSED** [review]

File: include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp (hunk 2 above)
Comment (zjing14) [reply include/ck/tensor_operation/gpu/element/unary_element_wise_operation.hpp:36]: Yes, these are experimental.

**Review by bartekxk — APPROVED** [review]

**Review by illsilin — APPROVED** [review]

**Pull request merged**